### PR TITLE
some corrections after integration tests

### PR DESCRIPTION
**Pull request by atsju**

### CODE REVIEW

File: BSP/sharpMemoryLCD/LS013B7DH03.c [hunk 1]
```diff
@@ -241,11 +246,26 @@ bool sharpMemoryLCD_printTextLine(uint8_t line, const char *text, uint8_t nbChar
                     // we need to update pixels inside a byte.
                     // modify the pixel/bit in the correct byte of the buffer
                     // by taking the correct bit in the correct byte of font table
-                    pixelBuf[y*SCREEN_WIDTH + screenX/NB_BIT_PER_BYTE] |= (1<<(screenX % NB_BIT_PER_BYTE) & Font16.table[indexCurrentByteInFont]);
+                    bool fontPixelBit = (1<<((NB_BIT_PER_BYTE-1)-(x % NB_BIT_PER_BYTE))) & Font16.table[indexCurrentByteInFont];
```
Comment (Hish15) [inline BSP/sharpMemoryLCD/LS013B7DH03.c]: This deserve intermediate var with clear names.
It is hard to grasp what is happening on these multiples instructions.

File: BSP/sharpMemoryLCD/LS013B7DH03.c [hunk 2]
```diff
@@ -241,11 +246,26 @@ bool sharpMemoryLCD_printTextLine(uint8_t line, const char *text, uint8_t nbChar
...
+                    if(fontPixelBit)
+                    {
+                        pixelBuf[(y*SCREEN_WIDTH + screenX)/NB_BIT_PER_BYTE] ^= 1<<((NB_BIT_PER_BYTE-1) - (screenX%NB_BIT_PER_BYTE));
```
Comment (Hish15) [inline BSP/sharpMemoryLCD/LS013B7DH03.c]: same here.

File: BSP/sharpMemoryLCD/LS013B7DH03.c [hunk 3]
```diff
@@ -50,6 +50,27 @@ static void LCDdisplayEnable(bool en)
...
+ * And everything needs to be rotated because screen is upside down.
+ *
+ * @param pixelBuf display buffer where one bit is one pixel.
+ * @param nbBytePixelAsciiLine length of buffer. This must be multiple of 2
+ */
+static void reorderBitsToScreen(uint8_t *pixelBuf, uint16_t nbBytePixelAsciiLine)
+{
+    //TODO this could be optimized doing 4 bytes at a time (with __REV instruction)
+    // check length is multiple of 2
+    assert_param((nbBytePixelAsciiLine%2) == 0);
```
Comment (Hish15) [inline BSP/sharpMemoryLCD/LS013B7DH03.c:67]: Does it make sense to make an if statement and do nothing if not multiple ?

File: BSP/sharpMemoryLCD/LS013B7DH03.c [hunk 4]
```diff
@@ -69,11 +90,14 @@ static bool LCDupdateDisplay(uint8_t screenLine, uint8_t *pixelBuf, uint16_t nbB
...
         LCDslaveSelect(true);
+        HAL_Delay(1); //TODO this delay could be much less, but need a delay_us function
         uint8_t cmd_buffer[2] = {CMD_DATA_UPDATE, 0x00};
         for(uint16_t i=0; i<(nbBytes*NB_BIT_PER_BYTE)/SCREEN_WIDTH ; i++)
         {
-            cmd_buffer[1] = screenLine+i;
+            // screen is mounted upside down
+            cmd_buffer[1] = SCREEN_HEIGHT-screenLine+i;
             // send "data update" command to correct line
             if(HAL_SPI_Transmit(&SpiHandle, cmd_buffer, sizeof(cmd_buffer), 1000) != HAL_OK)
```
Comment (Hish15) [inline BSP/sharpMemoryLCD/LS013B7DH03.c]: Just spoted this literal, is it a timeout?

File: BSP/sharpMemoryLCD/LS013B7DH03.c (hunk 3 above)
Comment (atsju) [reply BSP/sharpMemoryLCD/LS013B7DH03.c:67]: assert is enough I think. It's a sanity check but it's always the case when you print full lines. And this screen requires to print full lines anyway

**Review by Hish15 — APPROVED** [review]

**Pull request merged**